src/components/OSCALControlProse.js

```diff
@@ -102,6 +115,23 @@ function getConstraintsDisplay(modificationSetParameters, parameterId) {
   return constraintDescriptions.join("\n");
 }
 
+/**
+ * Retrieve the set-parameters when contained in the top-level implemented requirement statement
+ *
+ * @param {Object} implReqStatements statements on how the control is implemented
+ * @param {String} componentId uuid of the component and desired by-component
+ * @returns An array of set-parameters
+ */
+function getImplReqSetParameters(implReqStatements, componentId) {
+  // Get the top-level implemented requirement statement
+  const topLevelByComp = implReqStatements
+    ?.find((statement) => statement["statement-id"].endsWith("_smt"))
+    ?.["by-components"]?.find(
+      (byComp) => byComp["component-uuid"] === componentId
+    );
+  return topLevelByComp?.["set-parameters"] || null;
```

Comment:
```suggestion
  return implReqStatements
    ?.find((statement) => statement["statement-id"].endsWith("_smt"))
    ?.["by-components"]?.find(
      (byComp) => byComp["component-uuid"] === componentId
    )
    ?.["set-parameters"] || null;
```

Any reason to keep the temporary value variable there?